misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java

```diff
@@ -387,6 +388,14 @@ public FunctionBodyBlockNode transform(FunctionBodyBlockNode functionBodyBlockNo
         Token openBrace = formatToken(functionBodyBlockNode.openBraceToken(), 0, 1);
         indent(); // increase indentation for the statements to follow.
         NodeList<StatementNode> statements = formatNodeList(functionBodyBlockNode.statements(), 0, 1, 0, 1, true);
+        if (functionBodyBlockNode.namedWorkerDeclarator().isPresent()) {
+            NamedWorkerDeclarator namedWorkerDeclarator = formatNode(functionBodyBlockNode.namedWorkerDeclarator().get(),
+                    0, 2);
```

Comment:
Shall we keep a single trailing newline here?

Reply:
It was suggested by @SupunS to keep a newline between workers. 

https://github.com/ballerina-platform/ballerina-lang/pull/26339#discussion_r504374819

Reply:
I think he must have meant a single newline, let's get this clarified from @SupunS.

Reply:
I was initially thinking of a single newline. 
But on a second thought, I think we should go by the same rules as for the statements. That is, a minimum of a single newline, and a maximum of two newlines. WDYT?

btw, have we implemented the `transform(NamedWorkerDeclarator namedWorkerDeclarator)` method? We will have the same concern there as well?

Reply:
We should implement the `transform(NamedWorkerDeclarator namedWorkerDeclarator)` method in the new formatting tree modifier as well. 
Should we go with a single newline for worker blocks? 
Otherwise there will be an extra newline at the end of the last worker block as suggested by @IrushiL

Reply:
@sanjana Let's go in the same approach as in statements. It handles those cases as well.

Reply:
@SupunS Yeah, it's better to preserve an empty new line if the user has explicitly kept new lines.

Reply:
okay, I've fixed it.